src/js/elements/menu.js

```diff
@@ -1,174 +1,194 @@
-        }
+      let subindicators = detail.metadata.groups.filter((group) => group.name === detail.metadata.primary_group)[0];
+      this.addSubIndicators(childWrapper, category, subcategory, indicator, subindicators, indicators, detail.choropleth_method, detail.id);
     }
-
-    function addSubcategories(wrapper, category, subcategories) {
-        var newCategory = categoryTemplate.cloneNode(true)
-        $(newCategory).removeClass(hideondeployClsName);
-        $(".data-category__h1_title div", newCategory).text(category);
-        $('.' + loadingClsName).addClass('hidden');
-        $('.' + noDataWrapperClsName).addClass('hidden');
-        parentContainer.append(newCategory);
-        var h2Wrapper = $(".data-category__h1_wrapper", newCategory);
-        $(".data-category__h2", h2Wrapper).remove();
-
-        for (const [subcategory, detail] of Object.entries(subcategories)) {
-            let hasChildren = checkIfSubCategoryHasChildren(subcategory, detail);
-
-            if (hasChildren) {
-                let count = subindicatorsInSubCategory(detail);
-                if (count > 0) {
-                    addIndicators(h2Wrapper, category, subcategory, detail.indicators);
-                }
-            }
+  }
+
+
+  addSubcategories(wrapper, category, subcategories) {
+    var newCategory = categoryTemplate.cloneNode(true)
+    $(newCategory).removeClass(hideondeployClsName);
+    $(".data-category__h1_title div", newCategory).text(category);
+    $('.' + loadingClsName).addClass('hidden');
+    $('.' + noDataWrapperClsName).addClass('hidden');
+    parentContainer.append(newCategory);
+    var h2Wrapper = $(".data-category__h1_wrapper", newCategory);
+    $(".data-category__h2", h2Wrapper).remove();
+
+    for (const [subcategory, detail] of Object.entries(subcategories)) {
+      let hasChildren = this._model.checkIfSubCategoryHasChildren(detail);
+
+      if (hasChildren) {
+        let count = this._model.subindicatorsInSubCategory(detail);
+        if (count > 0) {
+          this.addIndicators(h2Wrapper, category, subcategory, detail.indicators);
         }
+      }
     }
+  }
 
-    function setActive(el) {
-        resetActive();
-        $(this).addClass("menu__link_h4--active")
-    }
+  setActive(el) {
+    this.resetActive();
+    $(this).addClass("menu__link_h4--active")
+  }
+
+  resetActive() {
+    $(".menu__link_h4--active", parentContainer).removeClass("menu__link_h4--active");
+  }
 
-    function resetActive() {
-        $(".menu__link_h4--active", parentContainer).removeClass("menu__link_h4--active");
-    }
 
+  main() {
     $(".data-menu__category").remove();
     let hasNoItems = true;
     let hiddenClass = hideondeployClsName;
     hiddenClass = 'hidden';
     $(parentContainer).find('.data-category').remove();
+    for (const [category, detail] of Object.entries(this._data)) {
+      let hasChildren = this._model.checkIfCategoryHasChildren(detail)
```

Comment:
use the model getter

Reply:
Hmmm, what advantage it there of calling the getter here. Shouldn't this be just fine?

Reply:
Done.